src/Learners/Classifier/BayesPointMachineClassifier.cs

```diff
@@ -328,10 +334,11 @@ public static class BayesPointMachineClassifier
         /// <typeparam name="TLabelSource">The type of a source of labels.</typeparam>
         /// <param name="fileName">The name of the file of a serialized binary Bayes point machine classifier.</param>
         /// <param name="mapping">The mapping used for accessing data in the native format.</param>
+        /// <param name="formatter">The formatter to use.</param>
         /// <returns>The binary Bayes point machine classifier instance.</returns>
         public static IBayesPointMachineClassifier<TInstanceSource, TInstance, TLabelSource, bool, Bernoulli, BayesPointMachineClassifierTrainingSettings, BinaryBayesPointMachineClassifierPredictionSettings<bool>>
             LoadBackwardCompatibleBinaryClassifier<TInstanceSource, TInstance, TLabelSource>(
-                string fileName, IBayesPointMachineClassifierMapping<TInstanceSource, TInstance, TLabelSource, bool> mapping)
+                string fileName, IBayesPointMachineClassifierMapping<TInstanceSource, TInstance, TLabelSource, bool> mapping, IFormatter formatter)
```

Comment:
> The whole point of backward compatibility is that we don't use formatters. Otherwise people could just use Load.

If you have a binary file the behaviour is that it uses the BinaryFileReader as the IReader. This currently uses BinaryFormatter internally, so there needs to be an IFormatter replacing it and you need to specify that formatter somehow?

Reply:
Actually it doesn't use BinaryFormatter internally, it just looks that way.

Reply:
You mean because it never actually calls ReadObject<T>()? (which uses BinaryFormatter internally?

Reply:
Yes